glean-core/python/glean/testing/__init__.py

```diff
@@ -36,14 +36,14 @@ def reset_glean(
     from glean import Glean
     from glean._dispatcher import Dispatcher
 
-    Dispatcher._testing_mode = True
-
     data_dir = None  # type: Optional[Path]
     if not clear_stores:
         Glean._destroy_data_dir = False
         data_dir = Glean._data_dir
 
     Glean._reset()
+    Dispatcher._testing_mode = True
```

Comment:
Why was this moved?

Reply:
`Glean._reset` waits for the worker thread to complete, and how that happens depends on the value of `_testing_mode`.  If `_testing_mode` changes before this happens, it won't wait on the worker thread, because it assumes all work is being done on the main thread.

I can add a comment to this effect here.